### PR TITLE
Added IsFinished, LogInfo and ClearOnFinish

### PR DESCRIPTION
**Pull request by SrLicht**
* Added a boolean flag to enable Log.Info messages in the code.

* Added a boolean flag to track whether the song has finished playing.

* Added a boolean flag to determine whether, upon finishing the track, the component and the referenceHub should be destroyed.

* Documented and improved PlayerIsConnected check to determine if the player is a DummyPlayer, as it's unnecessary to send audio packets to them.

* Added Visual Studio and Rider related build files to .gitignore.

### CODE REVIEW

**Comment by SrLicht** [comment]
@ced777ric PING. you're not going to ignore me for 3 months like #6

**Comment by ced777ric** [comment]
> @ced777ric PING. you're not going to ignore me for 3 months like #6

#6 was not ignored, it has conflicts and requested changes.

**Comment by SrLicht** [comment]
> > @ced777ric PING. you're not going to ignore me for 3 months like #6
> 
> #6 was not ignored, it has conflicts and requested changes.
github is trolling me or you forgot to hit request changes
imagen

**Comment by ced777ric** [comment]
image
Shows right there 🤷

File: SCPSLAudioApi/AudioCore/AudioPlayerBase.cs [hunk 1]
```diff
@@ -368,26 +404,37 @@ public virtual void Update()
...
+        /// <param name="hub">The ReferenceHub of the player to check.</param>
+        /// <returns>True if the player is fully connected and not a DummyPlayer; otherwise, false.</returns>
         private bool PlayerIsConnected(ReferenceHub hub)
         {
-            return hub.authManager.InstanceMode == ClientInstanceMode.ReadyClient && hub.nicknameSync.NickSet;
+            return hub.authManager.InstanceMode == ClientInstanceMode.ReadyClient &&
+                   hub.nicknameSync.NickSet && 
+                   !hub.isLocalPlayer &&
+                   !string.IsNullOrEmpty(hub.authManager.UserId) &&
+                   hub.authManager.UserId != null &&
```
Comment (ced777ric) [inline SCPSLAudioApi/AudioCore/AudioPlayerBase.cs]: The userid is checked for being null checked twice, once by IsNullOrEmpty and the second time with a null check.

**Review by ced777ric — CHANGES_REQUESTED** [review]

File: SCPSLAudioApi/AudioCore/AudioPlayerBase.cs (hunk 1 above)
Comment (SrLicht) [reply SCPSLAudioApi/AudioCore/AudioPlayerBase.cs]: ups

**Review by ced777ric — APPROVED** [review]

**Pull request merged**